### ISSUE TITLE
[🐛 Bug]: Cannot Interact with select lists inside an iframe - JSHandles can be evaluated only in the context they were created!

### ISSUE THREAD

**Issue by DJSdev** [webdriverio/webdriverio#7683]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

7.16.5

### Node.js Version

16.13.0

### Mode

WDIO Testrunner

### What happened?

This issue happens when using the 'devtools' protocol. Using the regular 'chromedriver' doesn't not reproduce this issue.

In my app, I have an iframe and inside of it is a `<select>` list with options. When I go to select the value using `.selectByVisibleText()`, `.selectByIndex()`, or `.selectByAttribute()` this will throw the error `JSHandles can be evaluated only in the context they were created!`.

I tested this code and it's only reproducible inside of an iframe. Running the same code on a select list outside of an iframe and there isn't any issue.

### What is your expected behavior?

The expected behavior is to be able to use `.selectByVisibleText()`, `.selectByIndex()`, and `.selectByAttribute()` while working inside of an iframe.

### How to reproduce the bug.

```js
// wdio.conf.ts
export const config: WebdriverIO.Config = {
  ...
  services: ['devtools'],
  ...
}
```

```js
describe('My Login application', () => {
  // This test fails
  it('With iframe', async () => {
    await browser.url(
      'https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_select'
    );
    const iframe = await $('#iframeResult');
    iframe.waitForExist();
    browser.switchToFrame(iframe);
    const selectList = await $('#cars');
    expect(await selectList.waitForDisplayed()).toEqual(true);
    await selectList.selectByVisibleText('Opel');
  });

  // This test passes
  it('Without iframe', async () => {
    await browser.url(
      'https://www.html.am/html-codes/forms/html-select-lists.cfm'
    );
    const selectList = await $('[name="fruit"]');
    expect(await selectList.waitForDisplayed()).toEqual(true);
    await selectList.selectByVisibleText('Lychee');
  });
});
```

### Relevant log output

```typescript
[0-0] 2021-11-10T21:38:58.977Z INFO devtools: COMMAND findElementFromElement("ELEMENT-5", "xpath", "./option[. = "Opel"]|./option[normalize-space(text()) = "Opel"]|./optgroup/option[. = "Opel"]|./optgroup/option[normalize-space(text()) = "Opel"]")
[0-0] 2021-11-10T21:38:58.981Z INFO devtools: RESULT { 'element-6066-11e4-a52e-4f735466cecf': 'ELEMENT-6' }
[0-0] 2021-11-10T21:38:58.982Z INFO devtools: COMMAND elementClick("ELEMENT-6")
[0-0] Error in "My Login application.With iframe"
JSHandles can be evaluated only in the context they were created!
Error: JSHandles can be evaluated only in the context they were created!
    at ExecutionContext.convertArgument (C:\Code\webdriveriotest\node_modules\puppeteer-core\src\common\ExecutionContext.ts:297:17)
    at Array.map (<anonymous>)
    at ExecutionContext._evaluateInternal (C:\Code\webdriveriotest\node_modules\puppeteer-core\src\common\ExecutionContext.ts:257:25)
    at ExecutionContext.evaluate (C:\Code\webdriveriotest\node_modules\puppeteer-core\src\common\ExecutionContext.ts:140:23)
    at ElementHandle.evaluate (C:\Code\webdriveriotest\node_modules\puppeteer-core\src\common\JSHandle.ts:161:42)
    at ElementHandle.$eval (C:\Code\webdriveriotest\node_modules\puppeteer-core\src\common\JSHandle.ts:921:40)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Element.wrappedCommand (C:\Code\webdriveriotest\node_modules\devtools\build\devtoolsdriver.js:102:26)
    at async Element.wrapCommandFn (C:\Code\webdriveriotest\node_modules\@wdio\utils\build\shim.js:131:29)
    at async Element.wrapCommandFn (C:\Code\webdriveriotest\node_modules\@wdio\utils\build\shim.js:131:29)
[0-0] 2021-11-10T21:38:59.012Z INFO devtools: COMMAND deleteSession()
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Duplicate of #6627

**Comment by eldrickb** [comment]
Sorry for the necro, but I'm new to WebdriverIO and was hoping to find some sort of explicit solution or workaround  for this issue. I cannot complete my project without being able to target select options in an iFrame, and nothing in this post or #6627 indicates a solution to me.

@christian-bromann's comment [here](https://github.com/webdriverio/webdriverio/issues/6627#issuecomment-809965265) describes "re-registering the network listener for the devtools package" yet I wasn't using the devtools package nor had it installed when I encountered this issue, nevermind interacting at all with the network listener. I doubt my solution is to download the devtools package and start to re-configure it, especially because I'm currently using wdio in standalone mode with next to no custom configuration.

Any help please? Can post my specific logs / code / etc and/or make my own post if need be, but OP has already described the problem and ways to reproduce it. Thanks.

**Comment by christian-bromann** [comment]
The workaround/solution to this is to fix the bug mentioned in #6627 or use a browser driver as this bug only affects devtools automation.

**Comment by hiChandravanshi** [comment]
I was facing the same issues while selecting a value in a dropdown that is inside an iframe. The following workaround worked for me 
- Executing the javascript to select the value
- Clicking twice on the dropdown after selecting the value
```
const iframeElement = await browser.$(iframeLocator);
await iframeElement.waitForDisplayed({ timeout: 30000 });
const iframeElementCss = await browser.findElement("css selector", iframeLocator);
await browser.switchToFrame(iframeElementCss);
await browser.execute((selector, valueToSet) => {
        document.querySelector(selector).value = valueToSet;
      }, ddlSelector, value);
const ddlElement = await browser.$(ddlSelector);
await ddlElement.click();
await ddlElement.click();
```
